pallets/parachain-staking/src/lib.rs

```diff
@@ -1832,6 +1832,21 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		fn delegator_bond_till_minimum(
+			delegator: &T::AccountId,
+			candidate: &T::AccountId,
+			minimum: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			Self::get_delegator_stakable_free_balance(delegator)
```

Comment:
Within the locking funds paradigm you need to know the key used to create the lock to determine free funds. So, it makes more sense to move this logic to parachain-staking. Maybe it should have been here in the first place.